jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/AccountResource.java

```diff
@@ -787,7 +790,15 @@ public Response payAllInvoices(@PathParam("accountId") final UUID accountId,
         final CallContext callContext = context.createCallContextWithAccountId(accountId, createdBy, reason, comment, request);
 
         final Account account = accountUserApi.getAccountById(accountId, callContext);
-        final Collection<Invoice> unpaidInvoices = invoiceApi.getUnpaidInvoicesByAccountId(account.getId(), clock.getUTCToday(), callContext);
+
+        LocalDate inputDate;
+        if(targetDate == null){
+            inputDate = clock.getUTCToday();
+        }
+        else{
+            inputDate = toLocalDate(targetDate);
+        }
+        final Collection<Invoice> unpaidInvoices = invoiceApi.getUnpaidInvoicesByAccountId(account.getId(), inputDate, callContext);
```

Comment:
Nit: formatting:
* Add space between words, so for example  `if` and `(` becomes `if (`, or `else` `{` becomes `else {`
* `else` is on the same line as `{`

Also, you could simplify this expression with following:
`final LocalDate inputDate = targetDate == null ? clock.getUTCToday() : toLocalDate(targetDate);`